compliance/compliance.go

```diff
@@ -100,6 +100,9 @@ type Test struct {
 	// RequiresImplicitReplace marks a test that requires the implementation of AFTOperation
 	// ADD for entries that already exist.
 	RequiresImplicitReplace bool
+	// RequiresNonDefaultNI marks a test that requires the implementation to configure
+	// entries for non-default network-instance.
+	RequiresNonDefaultNI bool
```

Comment:
nit: this seems too generic, how about: "RequiresNHInNonDefaultNI"? same with the flag.

Reply:
correction: "RequiresNHOnlyInDefault"

Reply:
Using the name "RequiresNonDefaultNINHG" to be consistent with "AddIPv4EntryDifferentNINHG"

By the way, although you suggested adding a "flush" version of AddIPv4EntryDifferentNINHG in private chat, I think there is some detail that needs to be fleshed out first. Can we defer that to a follow up PR?